prometheus/promhttp/http.go

```diff
@@ -408,6 +414,21 @@ type HandlerOpts struct {
 	// (which changes the identity of the resulting series on the Prometheus
 	// server).
 	EnableOpenMetrics bool
+	// EmitOpenMetricsTextCreatedSamples specifies if this handler should add, extra, synthetic
+	// Created Timestamps for counters, histograms and summaries, which for the current
+	// version of OpenMetrics are defined as extra series with the same name and "_created"
+	// suffix. See also the OpenMetrics specification for more details
+	// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1
+	//
+	// Created timestamps are used to improve the accuracy of reset detection,
+	// but the way it's designed in OpenMetrics 1.0 it also dramatically increases cardinality
+	// if the scraper does not handle those metrics correctly (converting to created timestamp
+	// instead of leaving those series as-is). New OpenMetrics versions might improve
+	// this situation.
+	//
+	// Prometheus introduced the feature flag 'created-timestamp-zero-ingestion'
+	// in version 2.50.0 to handle this situation.
+	EmitOpenMetricsTextCreatedSamples bool
```

Comment:
I feel that including `Emit` is a bit strong word because switching to OM v2 if  OM v2 has the metadata support for created timestamp AND someone set `EmitOpenMetricsTextCreatedSamples`  to true, they will no longer see the `_created` samples with 2.0 , only 1.0 - because we want to move CT into metadata. Which would mean we contradict the call to action `Emit!`.  Maybe calling it `EnableOpenMetricsTextCreatedSamples` would be slightly better to not contradict ourselves in 2.0.

Reply:
+1 cc @ArthurSens WDYT?

Reply:
Yep, sounds better!